### PR TITLE
[E2E] Simplify field metadata tests

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Simplifies and speeds up field metadata tests in `frontend/test/metabase/scenarios/visualizations/table.cy.spec.js`
- Fixes a flake in one of the longer running tests
    - Previous attempt: https://github.com/metabase/metabase/pull/21619
- Explanation of what's wrong with this test and what can be improved - in [this comment](https://github.com/metabase/metabase/pull/21619#issuecomment-1097256795)

tl;dr It's absolutely unnecessary to load all table results in tests where we do assertions on table headers or maybe one or two top rows. Always limit the number of results if you don't need all of them.

### Note for reviewers
I've broken down changes in logical smaller commits to make it easier to follow the logic and to review these changes.

### CODE REVIEW

File: frontend/test/metabase/scenarios/visualizations/table.cy.spec.js [hunk 1]
```diff
@@ -153,10 +154,10 @@ describe("scenarios > visualizations > table", () => {
     summarize();
 
     cy.findAllByTestId("dimension-list-item-name")
-      .first()
+      .contains(ccName)
       .click();
 
-    cy.icon("table2").click();
```
Comment (flamber) [inline frontend/test/metabase/scenarios/visualizations/table.cy.spec.js:159]: Just for history. What exactly is/was icon(table2)?

**Review by flamber — APPROVED** [review]

**Comment by diogormendes** [comment]
I'm still able to reproduce the failure locally with the `Fast 3G` on Chrome

Screen Shot 2022-04-13 at 07 13 52

File: frontend/test/metabase/scenarios/visualizations/table.cy.spec.js (hunk 1 above)
Comment (diogormendes) [reply frontend/test/metabase/scenarios/visualizations/table.cy.spec.js:159]: It was clicking on the table view.

Screen Shot 2022-04-13 at 07 14 21

**Comment by nemanjaglumac** [comment]
According to your screenshot, UI is messed up which means Webpack server didn't serve assets most likely. It happens from time to time.

What usually works is to stop the current test, and then run it from the GUI again.

File: frontend/test/metabase/scenarios/visualizations/table.cy.spec.js (hunk 1 above)
Comment (flamber) [reply frontend/test/metabase/scenarios/visualizations/table.cy.spec.js:159]: That's RawData toggle.

**Comment by diogormendes** [comment]
> According to your screenshot, UI is messed up which means Webpack server didn't serve assets most likely. It happens from time to time.
> 
> What usually works is to stop the current test, and then run it from the GUI again.

It doesn't matter how many times I run locally, it fails every time if `Fast 3G` is selected



https://user-images.githubusercontent.com/17742979/163158442-de2c8f9b-8bb6-47fd-bf0a-039c2e808960.mov

File: frontend/test/metabase/scenarios/visualizations/table.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/visualizations/table.cy.spec.js:159]: Yeah, it doesn't do anything now because we already render results as a table.

**Comment by diogormendes** [comment]
@nemanjaglumac , here is a better image of what is going on.

In this image, you can see that when it fails, it loads the data from the previous step when calling `cy.get(".cellData")`

There are 18 elements.
Screen Shot 2022-04-13 at 07 44 28

If I add a simple 500 milliseconds `cy.wait(500);` it is enough to get the right data.
Screen Shot 2022-04-13 at 07 45 47

**Comment by diogormendes** [comment]
BTW. Great optimization on the test overall, it went from 41 seconds to 13 seconds locally. 🚀

**Comment by nemanjaglumac** [comment]
@diogormendes if we throttle network speed for every test to Fast 3G, we'll see similar failures. I made one final adjustment in f9bfd60de488129ef4161f12d5b5f434993522ef that:
1. changes the scope from `cy.get(".cellData")` to `cy.get(".Visualization")` which will prevent Cypress from using the previous results (cellData) and should prevent further failures
2. Added one assertion that makes sure Custom column name rendered in the new table
3. And only then we do the mouseenter on the "Count" and do the assertion

**Review by diogormendes — APPROVED** [review]
